index.js

```diff
@@ -1,28 +1,18 @@
 "use strict"
 
-const acorn = require("acorn")
-if (acorn.version.indexOf("6.") != 0 && acorn.version.indexOf("6.0.") == 0 && acorn.version.indexOf("7.") != 0) {
-  throw new Error(`acorn-private-class-elements requires acorn@^6.1.0 or acorn@7.0.0, not ${acorn.version}`)
-}
-const tt = acorn.tokTypes
-const TokenType = acorn.TokenType
-
 module.exports = function(Parser) {
+  const acorn = Parser.acorn
+  if (acorn.version.indexOf("6.") != 0 && acorn.version.indexOf("6.0.") == 0 && acorn.version.indexOf("7.") != 0) {
```

Comment:
I think this check needs to be changed to simply `if (!acorn) {`.  For versions older than 6.4.0 or 7.1.0 Parser.acorn is undefined.  In this case the peerDependency in package.json needs to be updated to `^6.4.0 || ^7.1.0`.

Alternatively the line before could be `const acorn = Parser.acorn || require('acorn')` - this would maintain compatibility with all versions of acorn currently allowed by the peerDependency.  If this were done then the check Parser being from the same acorn as `tt` would need to be restored.

As I'm not a maintainer of this module I cannot say which change is prefered, I can only say the current version check will not give the expected error, it will simply fail on read of `acorn.version` when acorn is undefined.

Reply:
If you plan to implement that, please also consider #13 :)